network/src/peer/test.rs

```diff
@@ -380,7 +381,7 @@ fn peer_terminates_when_request_tx_has_dropped() {
     // Perform yamux handshake in a different task.
     let drop = async move {
         // Do not drop connection.
-        ManuallyDrop::new(connection);
+        forget(connection);
```

Comment:
@metajack @bmwill this is a tad interesting.

Reply:
Which lint did you hit with this? @phlip9 would have context on this part of the codebase

Reply:
Agghh I thought I purged all the `ManuallyDrop`... looks like I missed a spot.

```diff
@@ -365,15 +365,13 @@ fn peer_terminates_when_request_tx_has_dropped() {
     let (
         peer,
         peer_handle,
-        connection,
+        _connection,
         _peer_notifs_rx,
         _peer_rpc_notifs_rx,
         _peer_direct_send_notifs_rx,
     ) = build_test_peer(rt.handle().clone(), ConnectionOrigin::Inbound);
     // Perform yamux handshake in a different task.
     let drop = async move {
-        // Do not drop connection.
-        ManuallyDrop::new(connection);
         // Drop peer handle.
         drop(peer_handle);
     };

```

Reply:
This line wasn't a lint, but a warning I cleaned up at the same time as the other,
```
unused return value of `ManuallyDrop::<T>::new` that must be used
`#[warn(unused_must_use)]` on by default
if you don't need the wrapper, you can use `mem::forget` insteadrustc(unused_must_use)
```

Reply:
@phlip9 done, didn't know you could tell rust to ignore memory....